### ISSUE TITLE
Make a note read-only

### ISSUE THREAD

**Issue by BabakFarshchian** [Zettlr/Zettlr#4230]
### Checklist

- [X] Zettlr does not possess this feature (if it does, but the feature doesn't work out for you, propose an enhancement instead.)
- [ ] Zettlr's current features cannot be used to approximate this behaviour (otherwise, propose an enhancement instead.)
- [X] I have searched the documentation to make sure I did not overlook it.
- [ ] I have searched both open and closed issues to see if someone has already proposed this.

### Feature Proposal

This is mostly relevant for those of us working on manuscripts. When I have submitted a manuscript I would like to mark it as read-only (alternatively "archived"). It is easy to change notes in Zettlr (the app's strength!) but sometimes you just want to archive a note and not change it again.

### Example Workflow

* I am finished with a writing project (a manuscript, a blog, etc.), and sent it out for review or for publications.
* I go to an option (write-click/menu/keyword shortcut) and toggle the status of the file from writeable to read-only/archived.
* After this is done, some visual sign is turned on in some toolbar/note background color is changed, and I am not allowed to change that note again until I toggle its status again.

### Caveats

I am not sure how the editing can be prevented within Zettlr in the best way. 

### Do you wish to attempt implementing this yourself?

No

### Platforms

- [X] Windows
- [ ] macOS
- [ ] Linux

### Additional Information

Currently for doing this I go to the file system and make the file read-only. I have auto-save on in Zettlr. In this way Zettlr does not prevent me from editing the file, but it seems line autosave does not work and it does not warn me unless I try to close the file unsaved.

**Comment by nathanlesage** [comment]
Two thoughts on this request:

1. I won't implement a way to make files read-only; at least not yet.
2. You are absolutely right that Zettlr can – and therefore should – detect if a file is read-only and thus not allow you to edit it. The latter is something that is definitely important and worthwhile implementing.

**Comment by BabakFarshchian** [comment]
> I think he meant read only text, so one can go over his notes/links and relearn instead of typing. And no accidental changes happen. Just a button that makes typing cursor go away. Also no longer ctrl+click the links, just click. I am not a Zettlr user yet, just trying to learn.

Thank you @bfcns your suggestion also makes a lot of sense. My original idea was to make the file read-only. I did not think that the note could also become read-only in Zettlr only. This feature exists in Obsidian. It might be worth creating a new issue for it.

**Comment by nathanlesage** [comment]
I don't think we need a new issue. Making a note read only should be fairly straight forward with a simple button and a bit of state field logic. If someone wants to go ahead, please feel free to open a PR.

**Comment by Maziar123** [comment]
+1 preview mode as reader

**Comment by libook** [comment]
Zettlr offers excellent Markdown rendering, which makes it a favorite of mine for reading Markdown documents; however, being restricted to "edit mode" leaves me feeling uneasy—so much so that I dare not enable the auto-save feature.

I hope that one day there will be a button allowing me to toggle between edit mode and read-only mode at any time.
